### PR TITLE
Translate strings that appear in the cove web interface

### PR DESCRIPTION
**Pull request by Bjwebb**
Currently strings that only appear on the commandline, e.g. help messages are not translated.

### CODE REVIEW

**Comment by Bjwebb** [comment]
@jpmckinney flatten-tool is the only place from which strings appear in cove-oc4ids, that isn't a Django app. I've not been able to find a straightfoward way of dealing with that, as the language selected in Django is passed around differently to how gettext normally handles libraries (which uses env vars). I've written a short [i18n.py](https://github.com/OpenDataServices/flatten-tool/pull/362/files#diff-9f70dbe450808053ced7c0de7f33f81f35885422dccaf40e7ce4ce9ca4e8bdf1) to try to get the Django language if we're within Django, and otherwise have normal gettext behaviour. This works, but just feels a bit hacky, so I'd be interested to know if you have any thoughts on this?

**Comment by jpmckinney** [comment]
i18n.py seems reasonable – I'm not too familiar with patterns for translating CLIs.

File: flattentool/input.py [hunk 1]
```diff
@@ -200,7 +204,9 @@ def merge(base, mergee, debug_info=None):
                             + id_info
                         )
                     warn(
-                        'You may have a duplicate Identifier: We couldn\'t merge these rows with the {}: field "{}" in sheet "{}": one cell has the value: "{}", the other cell has the value: "{}"'.format(
+                        _(
+                            'You may have a duplicate Identifier: We couldn\'t merge these rows with the {}: field "{}" in sheet "{}": one cell has the value: "{}", the other cell has the value: "{}"'
+                        ).format(
                             id_info,
                             key,
                             debug_info.get("sheet_name"),
```
Comment (michaelwood) [inline flattentool/input.py:209]: I know this is quite a pain to change, but wrg to recent discussion having more context for translators might be good especially with multiple format strings. Perhaps where it is really non obvious from the en context use the %(blah)s method?

**Review by michaelwood — CHANGES_REQUESTED** [review]
Do think:
> I know this is quite a pain to change, but wrg to recent discussion having more context for translators might be good especially with multiple format strings. Perhaps where it is really non obvious from the en context use the %(blah)s method?

Should be resolved if we can, scanning down the po file output there are quite a few that would be hard to translate without the context provided by a var name.

**Review by michaelwood — APPROVED** [review]

File: flattentool/input.py (hunk 1 above)
Comment (Bjwebb) [reply flattentool/input.py:209]: We discussed this, and will open a new issue for this.

**Pull request merged**